wsmaster/che-core-api-factory-bitbucket-server/src/main/java/org/eclipse/che/api/factory/server/bitbucket/BitbucketServerPersonalAccessTokenFetcher.java

```diff
@@ -71,53 +73,42 @@ public BitbucketServerPersonalAccessTokenFetcher(
 
   @Override
   public PersonalAccessToken refreshPersonalAccessToken(Subject cheUser, String scmServerUrl)
-      throws ScmUnauthorizedException, ScmCommunicationException {
+      throws ScmUnauthorizedException, ScmCommunicationException, UnknownScmProviderException {
     // #fetchPersonalAccessToken does the same thing as #refreshPersonalAccessToken
-    return fetchOrRefreshPersonalAccessToken(cheUser, scmServerUrl);
+    return fetchPersonalAccessToken(cheUser, scmServerUrl);
```

Comment:
could you clarify why previously we fetched & refreshed the token? is the comment above still relevant

Reply:
Good catch, I added the refresh token logic so with that Bitbucket Server will support [refresh personal access token on workspace startup](https://eclipse.dev/che/docs/stable/administration-guide/configuring-oauth-for-git-providers/)